pkg/operator/events/recorder.go

```diff
@@ -0,0 +1,92 @@
+package events
+
+import (
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+type RecorderSourceFunc func() (RecorderSource, error)
```

Comment:
Why not use `k8s.io/client-go/tools/record` package instead of writing your own, we already vendor it

Reply:
Because there is no way to only use the EventRecorder without adding EventBroadcaster. Because the EventRecorder does not set all the events fields we want to set and I really want to be in control what fields are set and how. Because we will have to create an instance of this recorder everything we creating an event to fill the EventSource (as you can't dynamically change the involved object, etc..)

In summary, I'm not a fan of re-using things if they don't fit this use case. Especially if this recorder does not need to be super sophisticated and we don't really need/want the sophistication of EventSink.

Reply:
Discussed this in length on Slack, I'm ok with it for now :wink:

Reply:
How about instead having your own implementation of `EventRecorder` interface, this will save us some work in the future, if we need to switch?